### PR TITLE
Fix Bug in Resource Ref Verifier

### PR DESCRIPTION
**Pull request by chidozieononiwu**
- Fix bug in [eng/common/scripts/Verify-Resource-Ref.ps1](https://github.com/Azure/azure-sdk-tools/compare/main...chidozieononiwu:bugFix/ResourceRefVerifier?expand=1#diff-cc0dada0b261e5b5d0bd53f105529131abf0e67052ba76c3ef983c6bdeef644e)

### CODE REVIEW

**Comment by weshaggard** [comment]
Is there an issue for this or did you just spot it some other way?

File: eng/common/scripts/Verify-Resource-Ref.ps1 [hunk 1]
```diff
@@ -9,7 +9,7 @@ foreach ($file in $ymlfiles)
   $ymlContent = Get-Content $file.FullName -Raw
   $ymlObject = ConvertFrom-Yaml $ymlContent -Ordered
 
-  if ($ymlObject.Contains("resources"))
+  if (($null -ne $ymlObject) -and ($ymlObject.Contains("resources")))
```
Comment (weshaggard) [inline eng/common/scripts/Verify-Resource-Ref.ps1:12]:
```suggestion
  if (!$ymlObject -and $ymlObject.Contains("resources"))
```

Comment (benbp) [reply eng/common/scripts/Verify-Resource-Ref.ps1:12]: Wouldn't it be the opposite?

```suggestion
  if ($ymlObject -and $ymlObject.Contains("resources"))
```

Comment (benbp) [reply eng/common/scripts/Verify-Resource-Ref.ps1:12]: Also this script does not have strict mode, so I think you could also do:

```suggestion
  if ($ymlObject?.Contains("resources"))
```

Comment (weshaggard) [reply eng/common/scripts/Verify-Resource-Ref.ps1:12]: Yes talk @benbp's suggestion :)

Comment (chidozieononiwu) [reply eng/common/scripts/Verify-Resource-Ref.ps1:12]: I don't think @benbp suggestion works. At least from my local testing even after doing `Set-StrictMode -Off`

Comment (weshaggard) [reply eng/common/scripts/Verify-Resource-Ref.ps1:12]: That is a PS 7+ feature, so perhaps your PS version is older. Even with out that new feature his first suggestion is the correct one over mine.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**